starter_code/src/movies.js

```diff
@@ -1,19 +1,157 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+  let rate = [];
+  movies.forEach(function(element) {
+    rate.push(Number(element.rate));
+  });
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
+  const average =
+    rate.reduce(function(total, amount) {
+      return total + amount;
+    }, 0) / movies.length;
+  return parseFloat(average.toFixed(2));
+}
```

Comment:
It works perfectly fine! In order to write shorter code you could directly use the reduce method on the movies. And accessing the rate using the dot notation. 

e.g. 
function ratesAverage(movies) {
  let totalSum = movies.reduce(function(accumulator, value) {
    return accumulator + parseInt(value.rate, 10);
  }, 0);
  return totalSum / movies.length;
}